core/src/main/java/org/pentaho/di/core/database/Database.java

```diff
@@ -3435,7 +3435,7 @@ public void cleanupLogRecords( LogTableCoreInterface logTable ) throws KettleDat
       pstmt = connection.prepareStatement( databaseMeta.stripCR( sql ) );
 
       RowMetaInterface r = new RowMeta();
-      r.addValueMeta( new ValueMetaString( "TRANSNAME" ) );
+      r.addValueMeta( new ValueMetaString( "TRANSNAME", 255, -1 ) );
```

Comment:
Why hardcode 255 ? Shouldn't that come from somewhere else like the dialect ?

Reply:
https://github.com/pentaho/pentaho-kettle/blob/5099e34abeee5a604a3715078d8ff75fe3e11269/engine/src/main/java/org/pentaho/di/core/logging/TransLogTable.java#L202

TRANSNAME is always defined with 255 in our code. The issue is only with LOG_FIELD which is a TEXT / LONGTEXT and is different in each DB engine.